#! /bin/sh -e
## 21_sb-duart.dpatch by Thiemo Seufer <seufer@csv.ica.uni-stuttgart.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Export sb1250_duart_present

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p0 ${patch_opts} < $0;;
    -unpatch) patch -R -p0 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

--- drivers/char/Makefile.old	2004-05-08 09:54:05.000000000 +0200
+++ drivers/char/Makefile	2004-05-08 09:54:29.000000000 +0200
@@ -25,7 +25,7 @@ export-objs     :=	busmouse.o console.o 
 			misc.o pty.o random.o selection.o serial.o \
 			sonypi.o tty_io.o tty_ioctl.o generic_serial.o \
 			au1000_gpio.o vac-serial.o hp_psaux.o nvram.o \
-			scx200.o fetchop.o hydrogen3_buttons.o
+			scx200.o fetchop.o hydrogen3_buttons.o sb1250_duart.o
 
 mod-subdirs	:=	joystick ftape drm drm-4.0 pcmcia
 
--- drivers/char/sb1250_duart.c.old	2004-05-08 09:52:43.000000000 +0200
+++ drivers/char/sb1250_duart.c	2004-05-08 09:53:45.000000000 +0200
@@ -953,3 +953,7 @@ void __init sb1250_serial_console_init(v
 }
 
 #endif /* CONFIG_SIBYTE_SB1250_DUART_CONSOLE */
+
+EXPORT_SYMBOL(sb1250_duart_present);
+
+MODULE_LICENSE("GPL");
